src/main/cc/wfa/virtual_people/core/model/utils/distributed_consistent_hashing_test.cc

```diff
@@ -0,0 +1,250 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "wfa/virtual_people/core/model/utils/distributed_consistent_hashing.h"
+
+#include "absl/container/flat_hash_map.h"
+#include "absl/memory/memory.h"
+#include "absl/status/statusor.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "src/test/cc/testutil/matchers.h"
+#include "src/test/cc/testutil/status_macros.h"
+
+namespace wfa_virtual_people {
+namespace {
+
+constexpr int kSeedNumber = 10000;
+
+TEST(DistributedConsistentHashingTest, TestEmptyDistribution) {
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  EXPECT_THAT(
+      DistributedConsistentHashing::Build(std::move(distribution)).status(),
+      wfa::StatusIs(absl::StatusCode::kInvalidArgument, ""));
+}
+
+TEST(DistributedConsistentHashingTest, TestZeroProbabilitiesSum) {
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, 0);
+  distribution->emplace_back(1, 0);
+  EXPECT_THAT(
+      DistributedConsistentHashing::Build(std::move(distribution)).status(),
+      wfa::StatusIs(absl::StatusCode::kInvalidArgument, ""));
+}
+
+TEST(DistributedConsistentHashingTest, TestNegativeProbability) {
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, -1);
+  EXPECT_THAT(
+      DistributedConsistentHashing::Build(std::move(distribution)).status(),
+      wfa::StatusIs(absl::StatusCode::kInvalidArgument, ""));
+}
+
+TEST(DistributedConsistentHashingTest, TestOutputDistribution) {
+  // Distribution:
+  // choice_id probability
+  // 0         0.4
+  // 1         0.2
+  // 2         0.2
+  // 3         0.2
+  auto distribution = absl::make_unique<std::vector<DistributionChoice>>();
+  distribution->emplace_back(0, 0.4);
+  distribution->emplace_back(1, 0.2);
+  distribution->emplace_back(2, 0.2);
+  distribution->emplace_back(3, 0.2);
+  ASSERT_OK_AND_ASSIGN(
+      std::unique_ptr<DistributedConsistentHashing> hashing,
+      DistributedConsistentHashing::Build(std::move(distribution)));
+  absl::flat_hash_map<int32_t, int> output_counts = {
+      {0, 0},
+      {1, 0},
+      {2, 0},
+      {3, 0}};
+  for (int seed = 0; seed < kSeedNumber; ++seed) {
+    int32_t output = hashing->Hash(std::to_string(seed));
+    // The output should always be one of [0, 1, 2, 3].
+    EXPECT_TRUE(output_counts.find(output) != output_counts.end());
```

Comment:
Should this be `ASSERT_TRUE`?

Reply:
This is for testing that the output is one of the input choice ids.

Reply:
This is no longer required.